vpr/src/draw/draw.cpp

```diff
@@ -2955,19 +2963,83 @@ static void draw_pin_to_chan_edge(int pin_node, int chan_node, ezgl::renderer* g
 
     const t_grid_tile& grid_tile = device_ctx.grid[pin_rr.xlow()][pin_rr.ylow()];
     t_physical_tile_type_ptr grid_type = grid_tile.type;
-    VTR_ASSERT_MSG(grid_type->pinloc[grid_tile.width_offset][grid_tile.height_offset][pin_rr.side()][pin_rr.pin_num()],
-                   "Pin coordinates should match block type pin locations");
 
     float draw_pin_offset;
-    if (pin_rr.side() == TOP || pin_rr.side() == RIGHT) {
```

Comment:
I think this logic needs a comment.  Now that pin rr nodes have multiple edges, I don't think this logic is right.  I recommend you write a short comment explaining why you believe this is correct.

Reply:
Indeed. In the later part of this code block, I have logics to identify which side to be used for GUI. Therefore, I move the offset calculation there.

Reply:
@vaughnbetz No problem. I have added comments about why the positive/negative offset is needed